### PR TITLE
feat: build host iptables with nftables support

### PR DESCRIPTION
**Pull request by smira**
Build iptables with nftables support, and force to use nft version.

See https://github.com/siderolabs/talos/issues/9883

### CODE REVIEW

**Review by Unix4ever — APPROVED** [review]

File: iptables/pkg.yaml [hunk 1]
```diff
@@ -30,6 +33,19 @@ steps:
     install:
       - |
         make install DESTDIR=/rootfs
+      - |
+        # fix up symlinks which point to legacy version to point to nft version
```
Comment (dsseng) [inline iptables/pkg.yaml:37]: Is this necess

Comment (dsseng) [inline iptables/pkg.yaml:37]: Is this necessary? How other distros build this? Maybe patching the Makefile would be better?

**Review by dsseng — APPROVED** [review]

File: iptables/pkg.yaml (hunk 1 above)
Comment (smira) [reply iptables/pkg.yaml:37]: Our goal is not to have a generic solution which would work in any case. Talos enforces use of `nftables` (vs. legacy). Kubernetes does a wrapper which picks up iptables version by detecting existing rules in either of the backends. We know we do nft, so no reason to overcomplicate, and point default binaries to nft one. 

The host iptables should be used by CNI plugins, and they don't seem to be smart enough to figure out which version to use based on anything.

Comment (smira) [reply iptables/pkg.yaml:37]: and iptables Makefile is not configurable on that - it always does `iptables -> xtables-legacy-multi`.

This script should work even if the default is changed though.

Comment (smira) [reply iptables/pkg.yaml:37]:
```
/ # nsenter -t 1 -m /sbin/iptables -v
iptables v1.8.11 (nf_tables): no command specified
Try `iptables -h' or 'iptables --help' for more information.
```

Comment (smira) [reply iptables/pkg.yaml:37]: this is what a result looks like:

```
drwxr-xr-x 0/0               0 2019-06-02 01:34 sbin/
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/arptables -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/arptables-nft -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/arptables-nft-restore -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/arptables-nft-save -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/arptables-restore -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/arptables-save -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/arptables-translate -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ebtables -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ebtables-nft -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ebtables-nft-restore -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ebtables-nft-save -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ebtables-restore -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ebtables-save -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ebtables-translate -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-apply -> iptables-apply
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-legacy -> xtables-legacy-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-legacy-restore -> xtables-legacy-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-legacy-save -> xtables-legacy-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-nft -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-nft-restore -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-nft-save -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-restore -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-restore-translate -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-save -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/ip6tables-translate -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables -> xtables-nft-multi
-rwxr-xr-x 0/0            7052 2019-06-02 01:34 sbin/iptables-apply
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-legacy -> xtables-legacy-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-legacy-restore -> xtables-legacy-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-legacy-save -> xtables-legacy-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-nft -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-nft-restore -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-nft-save -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-restore -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-restore-translate -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-save -> xtables-nft-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/iptables-translate -> xtables-nft-multi
-rwxr-xr-x 0/0           79632 2019-06-02 01:34 sbin/xtables-legacy-multi
lrwxrwxrwx 0/0               0 2019-06-02 01:34 sbin/xtables-monitor -> xtables-nft-multi
-rwxr-xr-x 0/0          197856 2019-06-02 01:34 sbin/xtables-nft-multi
```

Comment (smira) [reply iptables/pkg.yaml:37]: so it basically changes "default" symlinks like `iptables` but `iptables-legacy` still points to legacy version

Comment (smira) [reply iptables/pkg.yaml:37]: I think we should probably drop iptables-legacy completely, for 1.10 with cgroups v1 (including kernel support).

Comment (dsseng) [reply iptables/pkg.yaml:37]: Yes, perhaps if no client complains. Should we add this to deprecation maybe?

Comment (smira) [reply iptables/pkg.yaml:37]: We never even documented that it exists, it should only be used by CNI plugins, and we want them to use nftables always.

**Pull request merged**